Fix/foreign lang connector bug

oscarssanchez opened the pull request:
Fixes #1051.

Hi @kasparsd ,

Could you use this PR that solves #1051 ?

This simply removes `sanitize_key()`. It looks like this value is later escaped:
https://github.com/xwp/stream/blob/develop/classes/class-list-table.php#L780 as well as label: https://github.com/xwp/stream/blob/develop/classes/class-list-table.php#L787

Since this is not getting saved into the database, my idea is it won't do any harm removing it, and there's no benefit adding an escaping function, which was my other solution. 

The unit tests seem to pass, as well as the problem is fixed by testing locally.

Thanks!

# Checklist

- [ ] Project documentation has been updated to reflect the changes in this pull request, if applicable.
- [x] I have tested the changes in the local development environment (see `contributing.md`).
- [x] I have added phpunit tests.


## Release Changelog

- Fix: Describe a bug fix included in this release.
- New: Describe a new feature in this release.


## Release Checklist

- [ ] This pull request is to the `master` branch.
- [ ] Release version follows [semantic versioning](https://semver.org). Does it include breaking changes?
- [ ] Update changelog in `readme.txt`.
- [ ] Bump version in `stream.php`.
- [ ] Bump `Stable tag` in `readme.txt`.
- [ ] Bump version in `classes/class-plugin.php`.
- [ ] Draft a release [on GitHub](https://github.com/xwp/stream/releases/new).


Change `[ ]` to `[x]` to mark the items as done.

kasparsd commented on connectors/class-connector-blogs.php:
```diff
@@ -74,7 +74,7 @@ public function get_context_labels() {
 
 			foreach ( $blogs as $blog ) {
 				$blog_details   = get_blog_details( $blog->blog_id );
-				$key            = sanitize_key( $blog_details->blogname );
```
Looks like context label keys have to be globally unique so how about we used something like `sprintf( 'blog-%d', $blog->blog_id );` for the key? That should be safe in every instance.

Would this break any of the existing records and the way they're filtered?

oscarssanchez replied:
You are right @kasparsd `blog_id` would be safer to use rather than `blog_name` !

oscarssanchez replied:
This didn't break existing records on my testing.

kasparsd commented:
@oscarssanchez Could you please also review the failing unit tests on single sites https://travis-ci.com/github/xwp/stream/jobs/324102448#L573? Thanks!

oscarssanchez commented:
Hi @kasparsd ,

Thanks for your feedback, I really appreciate you taking the time to review this, and sorry for the delay, as it has been busy weeks. 

I have updated the PR with your suggestion and updated the Unit Tests to reflect them. 

Thanks again!
